### PR TITLE
Resolve #105 connect to sources

### LINKED ISSUE #95: Stay consistent with variable names among modules
## What is the issue?
Follow naming convention when defining `tf` variables for Vault k/v

- related https://github.com/fredrikhgrelland/terraform-nomad-minio/issues/87

## Suggestion(s)/solution(s) [Optional]
**Use one naming convention**

```
vault_kv_field_<entity>
```
## Checklist (after created issue)
- [X] Added label(s)
- [ ] Added to project
- [x] Added to milestone

### LINKED ISSUE #102: Docker builds failing
## Current behaviour

https://github.com/Skatteetaten/vagrant-hashistack/issues/467

## Suggestion(s)/solution(s) [Optional]
Update to vagrant box 0.10 and template version 0.1.1

## Checklist (after created issue)
- [x] Added label(s)
- [ ] Added to project
- [x] Added to milestone

### LINKED ISSUE #105: Connect to other sources
## Feature description
Connect to other sources/connectors through the module "API"

## Why is it needed?
As of now, this modules seems to only support connecting to minio and hive connectors through the module "API". Connecting to other sources, e.g. postgresql, would be a nice feature and one we would use.

### LINKED ISSUE #111: Bump versions
## What is the issue?
Bump to `0.4.0` versions of:
- Postgres
- Minio
- Hive

## Checklist (after created issue)
- [x] Added label(s)
- [ ] Added to project
- [x] Added to milestone

### PR DESCRIPTION
**Pull request by claesgill**
[OLD]
⚠️ **This PR is a branch from the `presto-to-trino-include-plugin` not `master`. Merge pointing at that branch to see actuall changes (not Presto->Trino changes that are irrelevant). When we agree that this can be merged, i'll change to merge to master branch.**

[NEW]
ℹ️ **Updated PR to point to master branch.**

## Closes/fixes/resolves issue(s)?
Closes #105  
Closes #111  
Closes #95  
Closes #102  

## What was added/changed/fixed?
- Added config for Postgres connector
- Bumped versions of Postgres/Minio/Hive (0.4.0)

## Checklist (after created PR)
- [x] Added reviewers
- [x] Added assignee(s)
- [x] Added label(s)
- [ ] Added to project
- [x] Added to milestone

### CODE REVIEW

File: example/trino_standalone/main.tf [hunk 1]
```diff
@@ -65,10 +66,24 @@ module "trino" {
...
+    password      = module.postgres.password
+    database_name = module.postgres.database_name
+  }
+  postgres_vault_secret = {
+    use_vault_provider      = false
+    vault_kv_policy_name    = ""
+    vault_kv_path           = ""
+    vault_kv_field_username = ""
+    vault_kv_field_password = ""
+  }
```
Comment (zhenik) [inline example/trino_standalone/main.tf:82]: Just to clarify, in the current example do you (@claesgill ) use the same `postgres` instance for both `hive` and `trino`?

As I understood, you connect to the same `postgres` instance, to the same database that `hive` use and you use the same credentials that `hive` use. Am I right?

File: conf/nomad/trino_standalone.hcl [hunk 2]
```diff
@@ -163,15 +180,19 @@ job "${nomad_job_name}" {
...
         volumes = [
-          "local/trino/config.properties:/lib/trino/default/etc/config.properties",
-          "local/trino/catalog/hive.properties:/lib/trino/default/etc/catalog/hive.properties",
+          "local/trino/config.properties:/etc/trino/config.properties",
+          "local/trino/catalog/hive.properties:/etc/trino/catalog/hive.properties",
+          # Trino extra config volume destination
+          "local/trino/catalog/postgresql.properties:/etc/trino/catalog/postgresql.properties",
           # JVM settings. Memory GC etc.
-          "local/trino/jvm.config:/lib/trino/default/etc/jvm.config",
+          "local/trino/jvm.config:/etc/trino/jvm.config",
```
Comment (zhenik) [inline conf/nomad/trino_standalone.hcl:188]: @claesgill Why did you change `lib/` to `/etc`?

File: variables.tf [hunk 3]
```diff
@@ -205,3 +205,32 @@ variable "minio_vault_secret" {
...
+  })
+  description = "Set of properties to be able to fetch Postgres secrets from vault"
+  default = {
+    use_vault_provider      = false
+    vault_kv_policy_name    = "kv-secret"
+    vault_kv_path           = "secret/data/dev/trino"
+    vault_kv_field_username = "username"
+    vault_kv_field_password = "password"
+  }
+}
```
Comment (zhenik) [inline variables.tf:236]: Voices in my head:
Have you thought about making the whole `postgres` as a source - optional feature for the `current terraform-nomad-trino` module? 

We might need to support several sources, and the user will decide which feature does he/she needs to turn on.

**Review by zhenik — COMMENTED** [review]
Some comments

File: example/trino_standalone/main.tf (hunk 1 above)
Comment (claesgill) [reply example/trino_standalone/main.tf:82]: Yes, that's right. Should I add another Postgres? Or WDYT?

File: conf/nomad/trino_standalone.hcl (hunk 2 above)
Comment (claesgill) [reply conf/nomad/trino_standalone.hcl:188]: The old path did not work and Alexander enlightened me that the correct path should be `/etc/trino/` since `/lib/trino/default/etc/` is just a path "we" created. Putting any config in the old path does not seem to have any effect.

File: variables.tf (hunk 3 above)
Comment (claesgill) [reply variables.tf:236]: I agree, that should be something we need to plan for. As of now, the issue addressed support for Postgres module only. But I think we need to find a way of making this dynamical to support any service (at least regarding connectors).

File: example/trino_standalone/main.tf (hunk 1 above)
Comment (zhenik) [reply example/trino_standalone/main.tf:82]: I think at first iteration, I would be happy if you add this info to readme.md of example.

File: variables.tf (hunk 3 above)
Comment (zhenik) [reply variables.tf:236]: Could you register an issue for optional use of sources?

Comment (claesgill) [reply variables.tf:236]: Done: #112

File: example/trino_standalone/main.tf (hunk 1 above)
Comment (claesgill) [reply example/trino_standalone/main.tf:82]: Done: Updated the Readme example.

File: README.md [hunk 4]
```diff
@@ -197,6 +197,7 @@ module "trino" {
   source = "github.com/fredrikhgrelland/terraform-nomad-trino.git?ref=0.3.0"
 
   depends_on = [
+    module.postgres,
```
Comment (zhenik) [inline README.md:200]: Might be good to mention that for simplicity, the same `postgres` instance uses for both `hive` and `trino` (my initial thoughts)

**Review by oleptho — DISMISSED** [review]
LGTM

File: README.md (hunk 4 above)
Comment (claesgill) [reply README.md:200]: Will add!

Comment (claesgill) [reply README.md:203]: Done

**Review by zhenik — DISMISSED** [review]
LGTM

**Review by oleptho — APPROVED** [review]
LGTM

**Review by dangernil — APPROVED** [review]

File: example/trino_cluster/main.tf [hunk 5]
```diff
@@ -57,15 +57,31 @@ module "presto" {
...
+  postgres_vault_secret = {
+    use_vault_provider      = false
+    vault_kv_policy_name    = ""
+    vault_kv_path           = ""
+    vault_kv_field_username = ""
+    vault_kv_field_password = ""
+  }
 }
 
+# TODO: Update dependent modules version
```
Comment (zhenik) [inline example/trino_cluster/main.tf:83]: What exactly does it mean? Update which dependent modules? Update variables?

**Review by zhenik — CHANGES_REQUESTED** [review]

File: example/trino_cluster/main.tf (hunk 5 above)
Comment (claesgill) [reply example/trino_cluster/main.tf:83]: Dependent modules are postgres, minio etc. But I won't fix the trino_cluster in this version. I'll create a new issue regarding this since I believe it is a bigger task.

Comment (claesgill) [reply example/trino_cluster/main.tf:83]: #115

**Review by zhenik — APPROVED** [review]
